pkg/manager/namespace/manager.go

```diff
@@ -97,6 +98,18 @@ func (n *NamespaceManager) GetNamespace(nm string) (*Namespace, bool) {
 	return ns, ok
 }
 
+func (n *NamespaceManager) GetNamespaceByUser(user string) (*Namespace, bool) {
+	n.RLock()
+	defer n.RUnlock()
+
+	for _, ns := range n.nsm {
+		if ns.User() == user {
+			return ns, true
+		}
+	}
+	return nil, false
```

Comment:
I'm afraid distinguishing namespaces by usernames may never be applied. If you have listened to the share from the PM, creating different usernames may still be trouble for users.
For the dev tier, they are getting the tenant information from the user names. It's different with this way.

Reply:
But I think it's fine to support it now. We may remove it in the future if it's unncessary.

Reply:
We can go on development on routing while not apply any FD for official releases. But it is natural to develop: this PR merely has 100 LOC. I am still willing to do more refactors to make the code hackable and easier, both for dev and testing.

As for the dev tier, it is same, they are getting tenant information from user names, so does me. It is more of a problem of how to integrate GW and tiproxy.

Reply:
We can remove routing by user anytime and we can add it back anytime. I want such a framework instead of a fixed framework for "the current specialized product". So i added `getBackendIO` to give more flexibility. This it the first motivation.

Also PMs clearly did not have enough energy to cover all aspects of products. A flexible and high-performance GW is absolutely a possible HTAP solution. According to Liu Tang, the reason they rejected "one endpoint solution" is that it is hard to implement and unknown what to do. But from this PR, it may not be that hard and it can be very clear.